src/lib.rs

```diff
@@ -447,3 +449,132 @@ mod read;
 
 #[cfg(feature = "raw_value")]
 mod raw;
+
+/// Specifies how should bytes be (de)serialized
+///
+/// JSON does not natively support binary data. Protocols can specify their own
+/// mechanisms to handle binary data in JSON. Serde JSON supports different
+/// modes, see the details for each variant.
+///
+/// Note that the byte deserialization mode is only checked for types that are
+/// deserialized as bytes (when a type directly calls `deserialize_bytes` or
+/// `deserialize_byte_buf`). Types that are not self-describing (when a type
+/// calls `deserialize_any`) can't be deserialized as bytes.
+///
+/// The default mode is `IntegerArray`, which is the only format Serde JSON
+/// used to support.
+///
+/// The `Base64` mode is enabled with the `base64` crate feature.
+#[cfg(feature = "bytes_mode")]
+#[derive(Clone, Eq, PartialEq, Debug, Hash)]
+#[non_exhaustive]
+pub enum BytesMode {
```

Comment:
Not too happy about having this in `lib.rs`, but no existing module seems like a good place.